src/lib/operators/delete.hpp

```diff
@@ -26,12 +26,11 @@ class Delete : public AbstractReadWriteOperator {
       const std::shared_ptr<AbstractOperator>& copied_input_left,
       const std::shared_ptr<AbstractOperator>& copied_input_right) const override;
   void _on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) override;
-  void _on_commit_records(const CommitID cid) override;
+  void _on_commit_records(CommitID cid) override;
```

Comment:
The argument `CommitID` is of primitive type and will be copied by value. Therefore, marking it as `const` ist not relevant to function callers. This would be different though, if objects were passed by reference.
Using `const` in the function definition has an effect. It makes `cid` const within the implementation, which is a design choice.